### ISSUE TITLE
Working computer specs to run gear detector?

### ISSUE THREAD

**Issue by IamTechknow** [Team334/gear-detector#2]
Hi there, this is more of a general question, since your team got this working I figure it'd be better to post here. First of all this application of SSD Caffe is great. I'd been spending the last few days trying to get it to work myself but when I try to train the model it always get stuck. 

My computer spec is an Intel Core i7 4770k, GeForce 1060 GTX, and I have used an SSD and a HDD. I have tried building SSD Caffe on Ubuntu 17.10 (just came out last week) and 17.04, also I only used CUDA 9.0 and GCC 6.4 (CUDA 8.0 only works with GCC 5.3, but when I installed it I ran into errors with some STL library functions that prevent building so I don't bother with that anymore).

The end result with both versions of Ubuntu is that I can build Caffe, run the tests, and successfully train the LeNet example with my GPU. However whenever I try to build the SSD model (whether from the gear data, my own, or the example) it always results in two outcomes, either it crashes due to [issue 669](https://github.com/weiliu89/caffe/issues/669) or I comment out the line that can cause that crash, recompile, and get `Data layer prefetch queue empty` and the program just stalls. So basically I like to know is what version of Ubuntu you used, what version of CUDA and GCC, and any other steps you had to take to get this working. Thanks for reading!

**Comment by augustt198** [comment]
Yikes, that looks like a pretty tricky problem. Unfortunately I deleted the ubuntu installation I used to run this a while ago. What I remember is that it was pretty much a clean installation of 16.04, with CUDA 8 (not sure about gcc). I ran it on a GTX 1080.

When it comes to data loading problems my only idea is to mess with the batch size in ssd_pascal_steamworks.py (maybe you're trying to load too much data onto your gpu).

**Comment by augustt198** [comment]
Another idea: https://stackoverflow.com/questions/43199328/caffe-stick-in-data-layer-prefetch-queue-empty this guy says the caffe just couldn't find the data. Maybe there was a problem in making the LMDB files... the whole process is pretty sketchy.

**Comment by IamTechknow** [comment]
Thanks I'll keep that in mind. It turns out there is a suggestion to comment out the error preprocessor tags in the CAFFE wiki, so you likely just used GCC 6. I find the suggestion strange but oh well.

Fortunately I was able to solve the problem a day afterwards. Before I got this problem, I got an assertion failure of comparing two numbers in math_functions.cpp. At first I just commented it out, but instead I decided to swap the two numbers instead so that the check wouldn't fail. After that it work and I managed to train some models! I did have to adjust the batch size to a manageable amount, but not too low that the training would fail (mbox_loss increases rapidly to NaN). 

Thanks for the gear data too, I was able to train a model with it. We are working on our detection model for our FRC pre-season and will open source it eventually
